src/dotnet-serve/Program.cs

```diff
@@ -33,5 +47,166 @@ public static int Main(string[] args)
                 return 2;
             }
         }
+
+        static void ReadConfig(Config config, CommandLineOptions model)
+        {
+            // Port is special since it gets a value already even if none is provided,
+            // so we use the config value as an override instead.
+            if (config.TryGetNumber("serve", "port", out var port))
+            {
+                model.Port = (int)port;
+            }
+
+            if (model.Directory == null)
+            {
+                model.Directory = config.GetString("serve", "directory");
```

Comment:
Feedback for you on the Microsoft.DotNet.Config API...

I see every Get/Set method has to pass `section: "serve"`. It would be nice if the API had something similar to Microsoft.Extensions.Configuration which allows you to get an object that represents a section.
```c#
var cs = config.GetSection("serve");

model.Directory ??= cs.GetString("directory");
model. OpenBrowser ??= cs.GetString("open-browser");
// etc...
```

Reply:
Yep, adding an extension for Microsoft.Extensions.Configuration is in the pipeline. Created https://github.com/dotnetconfig/dotnet-config/issues/3 for tracking.

But a facade that "fixes" the section seems nice, yep.